File: Assignment Programs/Assignment24/program3.c
#include<stdio.h>

int FirstChar(char *str, char ch)
{
	int iCnt = 0;
	if(str == NULL)
	{
		return 0;
	}
	while(*str != '\0')
	{
		iCnt++;
		if(*str == ch)
		{
			break;
		}
		
		str++;
	}
	if(*str == '\0')
	{
		return 0;
	}
	else
	{
		return iCnt;
	}
} 
int main()
{
	char arr[20];
	char cValue = '\0';
	int iRet = 0;
	printf("Enter string\n");
	scanf("%[^'\n']s",arr);
	printf("Enter the character\n");
	scanf(" %c",&cValue);
	
	iRet = FirstChar(arr,cValue);
	printf("Character location is %d", iRet);
	return 0;
}

File: Assignment Programs/Assignment54/program3.java
import java.util.Scanner;

class MyArray
{
	public int Search(char arr[], char ch)
	{
		int iCnt = 0;
		for(int i=0;i<arr.length;i++)
		{
			if((ch >= 'A') && (ch <= 'Z'))
			{
				if((arr[i] == ch) || (arr[i] == ch+32))
				{
					iCnt++;
				}
			}
			else if((ch >= 'a') && (ch <= 'z'))
			{
				if((arr[i] == ch) || (arr[i] == ch-32))
				{
					iCnt++;
				}
			}
		}
		return iCnt;
	}
}

class program3
{
	public static void main(String args[])
	{
		int size = 0, iRet = 0;
		Scanner sobj = new Scanner(System.in);
		System.out.println("Enter the size of array");
		size = sobj.nextInt();
		
		char Arr[] = new char[size];
		System.out.println("Enter the elements of array");
		for(int i=0;i<size;i++)
		{
			Arr[i] = sobj.next().charAt(0);
		}
		System.out.println("Enter the character to search");
		char ch = sobj.next().charAt(0);
		MyArray mobj = new MyArray();
		iRet = mobj.Search(Arr,ch);
		System.out.println("Occurence of the character in array is "+iRet);
	}
}